mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/master/resourcecluster/ResourceCluster.java

```diff
@@ -82,6 +86,31 @@ public interface ResourceCluster extends ResourceClusterGateway {
      */
     CompletableFuture<Ack> refreshClusterScalerRuleSet();
 
+    /**
+     * Disables task executors that match the passed set of attributes
+     *
+     * @param attributes attributes that need to be present in the task executor's set of attributes.
+     * @param expiry     instant at which the request can be marked as complete. this is important because we cannot be constantly checking if new task executors match the disabled criteria or not.
+     * @return a future that completes when the underlying operation is registered by the system
+     */
+    CompletableFuture<Void> disableTaskExecutorsFor(Map<String, String> attributes, Instant expiry);
+
+    /**
+     * Gets the task executors to worker mapping for the given resource cluster
+     *
+     * @return a future mapping task executor IDs to the work they are doing
+     */
+    CompletableFuture<Map<TaskExecutorID, WorkerId>> getTaskExecutorWorkerMapping();
+
+    /**
+     * Gets the task executors to worker mapping for all task executors in the resource cluster that match the
+     * filtering criteria as represented by the attributes.
+     *
+     * @param attributes filtering criteria
+     * @return a future mapping task executor IDs to the work they are doing
+     */
+    CompletableFuture<Map<TaskExecutorID, WorkerId>> getTaskExecutorWorkerMapping(Map<String, String> attributes);
```

Comment:
Is there API to know how many/which executors are being disabled atm? Also does getResourceOverview has data reflected on disabled nodes.

Reply:
There's an API to get the information on how many executors are being disabled. I don't have an API to figure out which executors are disabled. I can add it though if you know the exact use-case.